src/roblox_api/instance.rs

```diff
@@ -91,6 +91,35 @@ impl LuaInstance {
         Ok(child)
     }
 
+    fn get_descendants(&self) -> rlua::Result<Vec<LuaInstance>> {
+        let tree = self.tree.lock().unwrap();
+
+        let instance = tree.get_by_ref(self.id).ok_or_else(|| {
+            rlua::Error::external("Cannot call GetDescendants() on a destroyed instance")
+        })?;
+
+        let mut descendants = Vec::new();
+        let mut stack = Vec::new();
```

Comment:
Which order does Roblox return descendants in? I would expect it to be breadth-first instead of depth-first. It'd be easy to switch us to breadth-first by using a `VecDeque` with `push_back` and `pop_front` here instead.

Reply:
Breadth-first, yeah.

Reply:
![image](https://user-images.githubusercontent.com/3190756/110455713-9db3b680-807d-11eb-89ec-3f27be40a90f.png)

Actually, looks like you're wrong.

Reply:
Depth first indeed! Let's lock that in.